pkg/azuredisk/azuredisk.go

```diff
@@ -344,30 +345,43 @@ func (d *Driver) CreateManagedDisk(ctx context.Context, options *ManagedDiskOpti
 	return diskID, nil
 }
 
-// TThe name must begin with a letter or number, end with a letter, number or underscore,
+// Disk name must begin with a letter or number, end with a letter, number or underscore,
 // and may contain only letters, numbers, underscores, periods, or hyphens.
 //
 // See https://docs.microsoft.com/en-us/rest/api/compute/disks/createorupdate#uri-parameters
-func getValidDiskName(volumeName string) string {
-	diskName := volumeName
-	if len(diskName) > diskNameMaxLength {
-		diskName = diskName[0:diskNameMaxLength]
-		klog.Warningf("since the maximum name length is 80, so it is truncated as (%q)", diskName)
+//
+//
+// Snapshot name must begin with a letter or number, end with a letter, number or underscore,
+// and may contain only letters, numbers, underscores, periods, or hyphens.
+// See https://docs.microsoft.com/en-us/rest/api/compute/snapshots/createorupdate#uri-parameters
+func getValidDiskOrSnapshotName(volumeName, volumeType string) string {
```

Comment:
keep `getValidDiskName` and add comment and link that snapshot naming rule is the same